src/Nethermind/Nethermind.Evm/Precompiles/ModExpPrecompile2565.cs

```diff
@@ -91,6 +104,53 @@ public long DataGasCost(byte[] inputData, IReleaseSpec releaseSpec)
             int expLength = (int)new UInt256(extendedInput.Slice(32, 32), true);
             int modulusLength = (int)new UInt256(extendedInput.Slice(64, 32), true);
 
+            return (baseLength, expLength, modulusLength);
+        }
+
+        public (byte[], bool) Run(byte[] inputData, IReleaseSpec releaseSpec)
+        {
+            Metrics.ModExpPrecompile++;
+
+            (int baseLength, int expLength, int modulusLength) = GetInputLengths(inputData);
+
+            byte[] modulusData = inputData.SliceWithZeroPaddingEmptyOnError(96 + baseLength + expLength, modulusLength);
+            using mpz_t modulusInt = ImportDataToGmp(modulusData);
+
+            if (gmp_lib.mpz_sgn(modulusInt) == 0)
+            {
+                return (new byte[modulusLength], true);
+            }
+
+            byte[] baseData = inputData.SliceWithZeroPaddingEmptyOnError(96, baseLength);
+            using mpz_t baseInt = ImportDataToGmp(baseData);
+            
+            byte[] expData = inputData.SliceWithZeroPaddingEmptyOnError(96 + baseLength, expLength);
+            using mpz_t expInt = ImportDataToGmp(expData);
+
+            using mpz_t powmResult = new();
+            gmp_lib.mpz_init(powmResult);
+            gmp_lib.mpz_powm(powmResult, baseInt, expInt, modulusInt);
+            
+            
+            using void_ptr data = gmp_lib.allocate((size_t) modulusLength);
+            ptr<size_t> countp = new(0);
+            gmp_lib.mpz_export(data, countp, 1, 1, 1, 0, powmResult);
+            int count = (int) countp.Value;
+
+
+            byte[] result = new byte[modulusLength];
+            Marshal.Copy(data.ToIntPtr(), result, modulusLength - count, count);
```

Comment:
don't we leak data without deallocating here?

Reply:
shall we deallocate in finally?

Reply:
using void_ptr data = gmp_lib.allocate((size_t) modulusLength);

we made data  disposable, then its deallocated